frontend/workflows/projectSelector/src/dash-hooks.tsx

```diff
@@ -22,6 +22,12 @@ export const useDashUpdater = (): useDashUpdaterReturn => {
   };
 };
 
-export const useDashState = (): DashState | undefined => {
-  return React.useContext<DashState | undefined>(DashStateContext);
+export const useDashState = (): DashState => {
+  const value = React.useContext<DashState | undefined>(DashStateContext);
```

Comment:
Should we just prevent this from ever being undefined? e.g. initialize the DashState in the context?

Reply:
It will only be undefined if you're not within the context. The DashState is initialized by the context since it's the Provider with a value.

Reply:
Right, I was thinking we change `DashStateContext`  to only have type `DashState` and set the initialization of the state on line 5 instead of using undefined.

Reply:
That would mask the instance where the context is uncontrolled by a provider (which is an invalid state). The context should be undefined if it's not initialized within a provider because the card won't work at all and something is wrong. Furthermore cards should not be updating and using the data of a global context outside of a provider if it were to have a default value. For further elaboration see here: https://www.carlrippon.com/react-context-with-typescript-p4/

This a common pattern used by many other large projects: https://sourcegraph.com/search?q=context:global+createContext.*%3F%5C%28undefined&patternType=regexp

Having a globally default value makes sense for simple immutable constant data (like a theme) but not for this type of complex mutable data IMO.